3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/adapters/registry/StatusGatorAdapter.scala

```diff
@@ -0,0 +1,126 @@
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package com.snowplowanalytics
+package snowplow
+package enrich
+package common
+package adapters
+package registry
+
+// Java
+import java.net.URI
+import org.apache.http.client.utils.URLEncodedUtils
+import org.apache.http.NameValuePair
+
+// Scala
+import scala.util.matching.Regex
+import scala.collection.JavaConversions._
+
+// Scalaz
+import scalaz._
+import Scalaz._
+
+// Jackson
+import com.fasterxml.jackson.core.JsonParseException
+
+// json4s
+import org.json4s._
+import org.json4s.JsonDSL._
+import org.json4s.jackson.JsonMethods._
+import org.json4s.scalaz.JsonScalaz._
+
+// Iglu
+import iglu.client.{
+  SchemaKey,
+  Resolver
+}
+
+// This project
+import loaders.CollectorPayload
+import utils.{JsonUtils => JU}
+
+/**
+ * Transforms a collector payload which conforms to
+ * a known version of the StatusGator Tracking webhook
+ * into raw events.
+ */
+object StatusGatorAdapter extends Adapter {
+
+  // Vendor name for Failure Message
+  private val VendorName = "StatusGator"
+
+  // Tracker version for an StatusGator Tracking webhook
+  private val TrackerVersion = "com.statusgator-v1"
+
+  // Expected content type for a request body
+  private val ContentType = "application/x-www-form-urlencoded"
+
+  // Schemas for reverse-engineering a Snowplow unstructured event
+  private val EventSchemaMap = Map (
+    "status_change" -> SchemaKey("com.statusgator", "status_change", "jsonschema", "1-0-0").toSchemaUri
+  )
+
+  /**
+   * Converts a CollectorPayload instance into raw events.
+   *
+   * A StatusGator Tracking payload contains one single event
+   * in the body of the payload, stored within a HTTP encoded
+   * string.
+   *
+   * @param payload The CollectorPayload containing one or more
+   *        raw events as collected by a Snowplow collector
+   * @param resolver (implicit) The Iglu resolver used for
+   *        schema lookup and validation. Not used
+   * @return a Validation boxing either a NEL of RawEvents on
+   *         Success, or a NEL of Failure Strings
+   */
+  def toRawEvents(payload: CollectorPayload)(implicit resolver: Resolver): ValidatedRawEvents =
+    (payload.body, payload.contentType) match {
+      case (None, _)                          => s"Request body is empty: no ${VendorName} events to process".failNel
+      case (_, None)                          => s"Request body provided but content type empty, expected ${ContentType} for ${VendorName}".failNel
+      case (_, Some(ct)) if ct != ContentType => s"Content type of ${ct} provided, expected ${ContentType} for ${VendorName}".failNel
+      case (Some(body),_)                     => {
+        if (body.isEmpty) {
+          s"${VendorName} event body is empty: nothing to process".failNel
+        } else {
+          val qsParams = toMap(payload.querystring)
+          try {
```

Comment:
here and for the others too, can't we move the try/catch closer to what can actually throw an exception, I'm guessing the parse?

Also can't this logic be factored out? I feel like I'm seeing the same logic in all the adapters